mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/schema/management/SearchSchemaCollector.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.schema.management;
+
+import java.util.Optional;
+
+import org.hibernate.search.engine.common.schema.management.SchemaExport;
+
+/**
+ * A schema collector that walks through all schema exports of a schema manager this collector is passed to.
+ */
+public interface SearchSchemaCollector {
+
+	/**
+	 * Called when an {@link SchemaExport index schema export} is encountered.
+	 *
+	 * @param backendName The name of the index's backend, or {@link Optional#empty()} for the default backend.
+	 * @param indexName The name of the index.
+	 * @param export The index schema export from a schema manager this collector is passed to.
+	 */
+	default void indexSchema(Optional<String> backendName, String indexName, SchemaExport export) {
+		// The default implementation that does nothing.
+		// Done so on purpose so when any other visiting methods are added to this interface users would still be
+		// able to only override/implement just those visitors that they are interested in.
```

Comment:
You can add the default implementation then? It's a backwards-compatible change.